server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -1099,39 +1076,41 @@ private IndexingStrategy(boolean currentNotFoundOrDeleted, boolean useLuceneUpda
             this.versionForIndexing = versionForIndexing;
             this.indexIntoLucene = indexIntoLucene;
             this.addStaleOpToLucene = addStaleOpToLucene;
+            this.seqNoOfNewerVersion = seqNoOfNewerVersion;
+            assert addStaleOpToLucene == false || seqNoOfNewerVersion >= 0 : "stale op [" + seqNoForIndexing + "] with invalid newer seqno";
```

Comment:
also assert it's higher than the seqNoForIndexing?

Reply:
also assert that if addStateOpToLucene is true, seqNoNewerVersion is unassigned? (move the assertion next to the other ones please)